mobx_codegen/lib/src/mobx_codegen_base.dart

```diff
@@ -20,17 +20,21 @@ import 'package:mobx_codegen/src/template/util.dart';
 import 'package:source_gen/source_gen.dart';
 
 class StoreGenerator extends Generator {
-  final _storeChecker = const TypeChecker.fromRuntime(Store);
-
   @override
   FutureOr<String> generate(LibraryReader library, BuildStep buildStep) =>
-      library.classes
-          .where(isValidStoreClass)
-          .expand((c) => expandToMixinCode(library, c))
-          .toSet()
-          .join('\n\n');
+      _generateCodeForLibrary(library).toSet().join('\n\n');
+
+  Iterable<String> _generateCodeForLibrary(LibraryReader library) sync* {
+    for (final classElement in library.classes) {
+      if (isValidMixinStoreClass(classElement)) {
+        yield* _generateCodeForMixinStore(library, classElement);
```

Comment:
What's the benefit of `yield*` instead of `yield` ?

Reply:
`yield*` yields each of the elements of an iterable individually

Without the *, the return type of this method would be `Iterable<Iterable<String>>`

Reply:
Ah ok. So it acts as an expansion operator. Cool. Learnt something new!